src/main/java/com/flightstats/hub/spoke/FileSpokeStore.java

```diff
@@ -178,14 +181,19 @@ private void keysInBucket(String key, OutputStream output) {
         }
         try {
             Collection<File> files;
-            if (resolution.equals("second")) {
+            if (Optional.ofNullable(resolution).orElse("FALSE").equals("second")) {
                 // filter all files in the minute folder that start with seconds
                 FileFilter fileFilter = new WildcardFileFilter(SpokePathUtil.second(key) + "*");
-                files = Arrays.asList(directory.listFiles(fileFilter));
+
+                files = Arrays.asList(Optional.ofNullable(directory.listFiles(fileFilter)).orElse(new File[]{}));
             } else {
                 files = FileUtils.listFiles(new File(path), null, true);
             }
-            for (File aFile : files) {
+            Collection<File> permanentFiles = files
+                    .stream()
+                    .filter(this::doesNotEndWith)
+                    .collect(Collectors.toList());
+            for (File aFile : permanentFiles) {
```

Comment:
Why not use `foreach` after the filter rather than putting it in an list and then looping the list?

Reply:
If you go this route (which I think you should), i'd extract the writing from to another method (`writeKeyFromFile`?)